### PR TITLE
Adding show notifications alert option

### PR DESCRIPTION
**Pull request by Shelvak**
I think that this will be helpful (at least for me :smile: ) so here we go.

[Feature] Add the possibility to show/alert about the new notifications count (and open Gh notifications page on notification click)

If enable, a not persistent alert is shown with the new notifications count.
Alert sample: 
captura de pantalla de 2018-08-29 21-22-21

Configs:
captura de pantalla de 2018-08-29 21-33-52

Sorry about the tab => space change, I have vim configured to automatically change this kind of things. To keep the review clean you can go to `Diff settings => Hide whitespace changes`

### CODE REVIEW

**Comment by alexduf** [comment]
I'll test it for a few days to see how it behaves

**Comment by Shelvak** [comment]
Awesome :+1:  let me know if you want to change any name or something =) . Cheers

File: extension.js [hunk 1]
```diff
@@ -220,9 +222,26 @@ const GithubNotifications = new Lang.Class({
...
     this.notifications = data;
     this.label && this.label.set_text('' + data.length);
     this.checkVisibility();
+    this.alertWithNotifications(lastNotificationsCount);
+  },
+
+  alertWithNotifications: function(lastCount) {
+    let newCount = this.notifications.length;
+
+    if (newCount && newCount != lastCount && this.showAlertNotification) {
```
Comment (alexduf) [inline extension.js]: So from my testing the first thing I noticed was that we get notified even if we're actively deleting stuff. If I have 2 notifications and I'm reading a PR (bringing it down to 1) I'll get notified.

What would you think about having something like that?
```javascript
if (newCount && newCount > lastCount && this.showAlertNotification) {
```

Comment (Shelvak) [reply extension.js]: That's a great point. Let me change that =)

**Comment by alexduf** [comment]
I'm not sure why but I get double notification from time to time. I'm not sure if it's related to your change or if that change is showing some gremlin hiding in the code.

Have you noticed it?

**Comment by Shelvak** [comment]
Hey Alex, yeah I noticed that when you're "not using" the pc the "dynamic notifications" are enqueued. That sucks. 
I can change the code to use a specific notification and check if you already have one, update it. And not notify again. What do you think?

**Comment by Shelvak** [comment]
Ah in this case I don't know what happened to you... I saw the multiple notifications only when I use different computers (all using the extension). When I come back to my home I see a few notifications.

**Comment by Shelvak** [comment]
@alexduf hey sorry for the delay. I improve the notifications to not repeat them. 
- If we don't have any unreaded notifications => the notification is created.
- If we have unreaded notifications => updated the created notification.
- [Improve] Click on the notification opens the Gh notifications page :tada: (I love that :smile:  )

**Comment by alexduf** [comment]
Great work! thanks again. I'll update my local version to your latest commit, test again for a few days and merge. (If I forget, ping me here :) )

**Comment by alexduf** [comment]
I didn't get the duplicate notification anymore, weird! Might have been arch-linux specific and got fixed. Since I have no remaining objection, I'll merge and release.

@Shelvak thanks again for your contribution, that's much appreciated!

**Comment by Shelvak** [comment]
Amazing Alex, happy to contribute =D  (And I learnt a lot about extensions)

**Comment by alexduf** [comment]
I've put the extension for review 13 days ago and it still hasn't been reviewed. I sent a gentle ping to see if anyone could review it.
So I haven't forgotten you, they probably are swamped :)

**Pull request merged**